Source/Scene/ImplicitTileCoordinates.js

```diff
@@ -136,20 +177,194 @@ Object.defineProperties(ImplicitTileCoordinates.prototype, {
       return MortonOrder.encode2D(this.x, this.y);
     },
   },
+
+  /**
+   * Get the tile index by adding the Morton index to the level offset
+   *
+   * @type {Number}
+   * @readonly
+   * @private
+   */
+  tileIndex: {
+    get: function () {
+      var levelOffset =
+        this.subdivisionScheme === ImplicitSubdivisionScheme.OCTREE
+          ? // (8^N - 1) / (8-1)
+            ((1 << (3 * this.level)) - 1) / 7
+          : // (4^N - 1) / (4-1)
+            ((1 << (2 * this.level)) - 1) / 3;
+
+      var mortonIndex = this.mortonIndex;
+      return levelOffset + mortonIndex;
+    },
+  },
 });
 
 /**
- * Given the (level, x, y, [z]) coordinates of the parent, compute the
- * coordinates of the child.
+ * Check that the two coordinates are compatible
+ * @param {ImplicitTileCoordinates} a
+ * @param {ImplicitTileCoordinates} b
+ * @private
+ */
+function checkMatchingProperties(a, b) {
```

Comment:
this name sounds more general than it is. Maybe `checkMatchingSubtreeShape(a, b)`?

The way I think of it is the subdivision scheme is like the "width" of the subtree and subtreeLevels is the "height"

I feel like we're going to get a theory of tree geometry by the time we merge this PR :laughing: